source/documentation/tools/jupyterlab/package-management.md

```diff
@@ -84,4 +85,92 @@ Before you can run this project, you need to install some Python packages using
 
     # install the python packages required
     . venv/bin/activate
-    pip install -r requirements.txt
\ No newline at end of file
+    pip install -r requirements.txt
+```
+
+## Poetry
+We advise the use of [poetry](https://python-poetry.org/docs/) as a library.
+
+### why use poetry 
+poetry stops conflicts between libraries and their dependencies.
+It makes sure that libraries play well together.
+If virtualenv is used without poetry, you may start experiencing library version conflict and start getting warnings.
+
+### setup steps:-
+- after creating your virtualenv, activate your venv 
```

Comment:
I think `poetry` handles virtual environments itself. If you do `poetry add ...` outside an existing venv it will create one that you can then access with `poetry shell`. Not sure how well this works with notebooks though, but I assume the existing documentation would stand.

Reply:
yes this is correct but it think poetry (or any other package manager) should be used in conjunction with the familiar (pip).

We can remove this step, just wanted to make sure that we have "single sources of truth".
- requirements.txt
- 1 venv that the user is familiar with